src/transport_ofi.c

```diff
@@ -1246,6 +1278,11 @@ int shmem_transport_fini(void)
 
     shmem_transport_ctx_destroy(&shmem_transport_ctx_default);
 
+    if (shmem_transport_ofi_target_ep &&
+        fi_close(&shmem_transport_ofi_target_ep->fid)) {
+        RAISE_ERROR_MSG("target endpoint close failed (%s)\n", fi_strerror(errno));
+    }
```

Comment:
This conditional should be unnecessary, since the target EP should always exist.

Reply:
Yeah, I was just keeping with the style of the conditional checks below.  Should I remove those too?  If I'm not mistaken, all these objects should exist?

Reply:
If they are for objects that should always exist, we should remove them.  Or potentially add an "else" case to act as an error check.